anta/tests/interfaces.py

```diff
@@ -163,22 +170,23 @@ def test(self) -> None:
         intf_not_configured = []
         intf_wrong_state = []
 
-        for interface_status in self.inputs.interfaces:
-            intf_status = get_value(command_output["interfaceDescriptions"], interface_status.interface, separator=";")
-            if intf_status is None:
-                intf_not_configured.append(interface_status.interface)
+        for interface in self.inputs.interfaces:
+            if (intf_status := get_value(command_output["interfaceDescriptions"], interface.name, separator="..")) is None:
+                intf_not_configured.append(interface.name)
                 continue
 
-            proto = intf_status["lineProtocolStatus"]
-            status = intf_status["interfaceStatus"]
+            status = "up" if intf_status["interfaceStatus"] in {"up", "connected"} else intf_status["interfaceStatus"]
+            proto = "up" if intf_status["lineProtocolStatus"] in {"up", "connected"} else intf_status["lineProtocolStatus"]
+
+            # If line protocol status is provided, prioritize checking against both status and line protocol status
+            if interface.line_protocol_status:
+                if interface.status != status or interface.line_protocol_status != proto:
+                    intf_wrong_state.append(f"{interface.name} is {status}/{proto}")
 
-            if interface_status.state == "up" and not (re.match(r"connected|up", proto) and re.match(r"connected|up", status)):
-                intf_wrong_state.append(f"{interface_status.interface} is {proto}/{status} expected {interface_status.protocol_status}/{interface_status.state}")
-            elif interface_status.state == "adminDown":
-                if interface_status.protocol_status == "up" and not (re.match(r"up", proto) and re.match(r"adminDown", status)):
-                    intf_wrong_state.append(f"{interface_status.interface} is {proto}/{status} expected {interface_status.protocol_status}/{interface_status.state}")
-                elif interface_status.protocol_status == "down" and not (re.match(r"down", proto) and re.match(r"adminDown", status)):
-                    intf_wrong_state.append(f"{interface_status.interface} is {proto}/{status} expected {interface_status.protocol_status}/{interface_status.state}")
+            # If line protocol status is not provided and interface status is "up", expect both status and proto to be "up"
```

Comment:
Please describe this behavior in the test docstring for our autogenerated documentation!